client/client.go

```diff
@@ -175,19 +175,7 @@ func (c *Client) downloadRoot() error {
 	var s *data.Signed
 	var raw []byte
 	if download {
-		logrus.Debug("downloading new root")
-		raw, err = c.remote.GetMeta(role, size)
-		if err != nil {
-			return err
-		}
-		hash := sha256.Sum256(raw)
-		if expectedSha256 != nil && !bytes.Equal(hash[:], expectedSha256) {
```

Comment:
Where did the ```expectedSha256 != nil``` case disappear to?

If it can happen, now it will raise an error (because bytes.Equal treats nil as an empty slice), which seems unwarranted.

If it cannot happen, other parts of this function are necessarily dead code and should be replaced by an assert or something.